Cabal/doc/nix-local-build.rst

```diff
@@ -462,6 +462,7 @@ a script that looks like:
     {- cabal:
     build-depends: base ^>= 4.11
                 , shelly ^>= 1.8.1
+    with-compiler: ghc-8.8.1
```

Comment:
This makes it sound like with-compiler is mandatory now. Might it be better to just have another paragraph here that mentions which fields are allowed?